packages/map/src/CdcMap.tsx

```diff
@@ -1449,6 +1448,16 @@ const CdcMap = ({ className, config, navigationHandler: customNavigationHandler,
       filters = generateRuntimeFilters(state, hashFilters, runtimeFilters)
 
       if (filters) {
+        if(JSON.stringify(runtimeFilters) !== JSON.stringify(filters)){
+          const urlParams = new URLSearchParams(window.location.search);
+          filters.forEach((filter, index) => {
+            if(filter.setByQueryParameter && urlParams.get(filter.setByQueryParameter)){
```

Comment:
This logic seems to repeat a lot. Could it be abstracted to a function and added to core?

Reply:
Moved to a shared function and addressed in latest commit